core/src/main/scala/org/broadinstitute/dsde/rawls/billing/BillingProjectOrchestrator.scala

```diff
@@ -130,20 +133,67 @@ class BillingProjectOrchestrator(ctx: RawlsRequestContext,
         )
       )
     } yield {}
+
+  def deleteBillingProjectV2(projectName: RawlsBillingProjectName): Future[Unit] =
+    for {
+      _ <- samDAO
+        .userHasAction(SamResourceTypeNames.billingProject,
+                       projectName.value,
+                       SamBillingProjectActions.deleteBillingProject,
+                       ctx
+        )
+        .flatMap {
+          case true => Future.successful()
+          case false =>
+            Future.failed(
+              new RawlsExceptionWithErrorReport(
+                errorReport = ErrorReport(StatusCodes.Forbidden, "You must be a project owner.")
+              )
+            )
+        }
+      billingProfileId <- billingRepository.getBillingProfileId(projectName)
+      projectLifecycle = billingProfileId match {
+        case None    => googleBillingProjectLifecycle
+        case Some(_) => bpmBillingProjectLifecycle
+      }
+      _ <- billingRepository.failUnlessHasNoWorkspaces(projectName)
+      _ <- projectLifecycle.preDeletionSteps(projectName, ctx)
+      _ <- unregisterBillingProjectWithUserInfo(projectName, ctx.userInfo)
+    } yield {}
+
+  // This code also lives in UserService.
+  def unregisterBillingProjectWithUserInfo(projectName: RawlsBillingProjectName,
```

Comment:
Is it worth trying to share this code? Currently this file does not have a dependency on UserService, and I would like to avoid introducing that.

Reply:
Seems reasonable as is to me. The logic is basically just doing two things with separate resources - not like it's super complex.